pkgs/build-support/node/fetch-npm-deps/src/parse/mod.rs

```diff
@@ -25,6 +29,10 @@ pub fn lockfile(content: &str, force_git_deps: bool) -> anyhow::Result<Vec<Packa
         })
         .collect::<anyhow::Result<Vec<_>>>()?;
 
+    if packages.is_empty() && !force_empty_cache {
+        bail!("No cacheable dependencies were found. Please inspect the upstream `package-lock.json` file and ensure that all required remote dependencies have `resolved` URLs and `integrity` hashes. If this is intentional and you would like to generate an empty cache anyways, set `forceEmptyCache = true`.");
```

Comment:
maybe a note here, what to do when not every remote dependency has `resolved` and `integrity` fields, would be good?

Reply:
Not every dependency needs to have it. This would be the case if _none_ of them do and it would therefore generate an empty cache

I'll make the comment clearer in a bit of the condition and what to do about it, thank you!

Reply:
Done! And I added a test too